elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala

```diff
@@ -117,6 +117,45 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
       }
     }
 
+    "Be able to delete a document that exists" in {
+      indexDocs(Seq(Document("doc1", Map("text" -> "here"))))
+      val resFut = restClient.query(index, tpe, QueryRoot(TermQuery("text", "here"),
+        sortOpt = Some(Seq(("_timestamp", "desc"))), timeout = Some(10)))
+
+      var foundDoc: Option[ElasticJsonDocument] = None
+      whenReady(resFut){res =>
```

Comment:
when ready returns the result of the function. You can do:
```
val foundDoc: ElasticJsonDocument = whenReady(resFut) { res =>
  res.raw.....
}
```

This also simplifies the subsequent code.